### ISSUE TITLE
Error response from daemon: reference does not exist

### ISSUE THREAD

**Issue by ameyaagashe** [docker/docker#25921]
**Output of `docker version`:**

```
[jim@docker1 ~]$ docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64
[jim@docker1 ~]$ 
```
**Output of `docker info`:**
```
[jim@docker1 ~]$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 1.12.0
Storage Driver: devicemapper
 Pool Name: docker-253:0-201697694-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 460.5 MB
 Data Space Total: 107.4 GB
 Data Space Available: 46.21 GB
 Metadata Space Used: 782.3 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay null host bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.22.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 2.781 GiB
Name: docker1
ID: IOFR:T2GH:WZN3:F3AQ:TTEJ:CAM2:D2SI:K4L6:4CFX:TU2H:BP6B:ATKD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
[jim@docker1 ~]$ 
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
1. Virtualbox
Windows 10 as HostOS. CentsOS 7.2 as Guest OS

**Steps to reproduce the issue:**
[jim@docker1 ~]$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
centos              latest              50dae1ee8677        4 weeks ago         196.7 MB
centos              centos6             cf2c3ece5e41        7 weeks ago         194.6 MB
[jim@docker1 ~]$ docker rmi 5
Error response from daemon: reference does not exist
[jim@docker1 ~]$ docker rmi c
Error response from daemon: reference does not exist
[jim@docker1 ~]$ docker rmi centos:latest
Error response from daemon: No such image: centos:latest
[jim@docker1 ~]$ docker rmi centos:centos6
Error response from daemon: No such image: centos:centos6
[jim@docker1 ~]$ 
**Describe the results you received:**
Error response from daemon: reference does not exist

**Describe the results you expected:**
docker images should be deleted.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by SQUIDwarrior** [comment]
I'm getting the same error with Docker Engine 1.12.1. Is there a way to completely blow away my image registry?

**Comment by admon** [comment]
rm -rf /var/lib/docker/image/devicemapper/imagedb/*/sha256/50dae1ee8677*

**Comment by thaJeztah** [comment]
@ameyaagashe are you still having this issue? Could it be you previously had the Red Hat fork of docker installed on that machine? If so, this could be related, as the Red Hat fork has modifications in this area that stores images using a non-standard reference.

**Comment by snout1979** [comment]
@thaJeztah I can confirm this occurs on some (not all) images when going from the Red Hat fork to the docker repo on Centos 7.

**Comment by thaJeztah** [comment]
I suspect this is due to switching indeed; unfortunately, I don't think there's anything we can fix this, as file-paths of images stored by the Red Hat fork can be incompatible with the official Docker binaries. Removing individual images (as @admon showed) _could_ work, or wiping `/var/lib/docker` to remove all images/containers.

I'll close this issue because of the above, but feel free to comment if you think additional changes are required

**Comment by bertalanimre** [comment]
I can confirm the exact same error and workaround for Fedora 24. Seems this issue effect all RHEL based distro?

**Comment by thaJeztah** [comment]
@bertalanimre if you also switched from the Red Hat fork to the official packages, then that's likely the same issue, yes

**Comment by GongT** [comment]
docker version:
```
[root@localhost gfw]# cat /etc/yum.repos.d/docker.repo
[dockerrepo]
name=Docker Repository
baseurl=https://yum.dockerproject.org/repo/main/fedora/$releasever/
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg
[root@localhost gfw]# dnf list installed docker-engine
Last metadata expiration check: 0:17:21 ago on Thu Feb  9 15:08:44 2017.
Installed Packages
docker-engine.x86_64                 1.13.1-1.fc24                 @dockerrepo
[root@localhost gfw]# docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:41:33 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:41:33 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

docker info:
```
[root@localhost gfw]# docker info
Containers: 8
 Running: 2
 Paused: 0
 Stopped: 6
Images: 21
Server Version: 1.13.1
Storage Driver: devicemapper
 Pool Name: docker-docker--pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 6.349 GB
 Data Space Total: 42.9 GB
 Data Space Available: 36.55 GB
 Metadata Space Used: 1.348 MB
 Metadata Space Total: 109.1 MB
 Metadata Space Available: 107.7 MB
 Thin Pool Minimum Free Space: 4.29 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.122 (2016-04-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.5-100.fc24.x86_64
Operating System: Fedora 24 (Twenty Four)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.778 GiB
Name: localhost.vm
ID: DJJT:4CSJ:ARKC:JQHU:DE2Z:6Z4S:CKP2:RVQD:KQSI:BJWS:LABC:ORFI
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

the test:
```
[root@localhost basic]# docker images
REPOSITORY             TAG                 IMAGE ID            CREATED             SIZE
xtaci/kcptun           latest              c0bc7b2644b8        5 days ago          305 MB
vimagick/shadowsocks   latest              6f48bd7bcdc0        7 days ago          54.6 MB
node                   alpine              0895ecd79009        7 days ago          54.6 MB
memcached              alpine              bfdef622d405        3 weeks ago         5.12 MB
alpine                 latest              88e169ea8f46        6 weeks ago         3.98 MB
fedora                 latest              a1e614f0f30e        7 weeks ago         197 MB
memcached              <none>              68fdd289dcd9        3 months ago        5.89 MB
ubuntu                 latest              c73a085dc378        4 months ago        127 MB
andyshinn/dnsmasq      latest              e6400cb52bf1        4 months ago        5.13 MB
mongo                  2.6                 9764c3ca91b2        4 months ago        391 MB
mongo                  latest              48b8b08dca4d        5 months ago        366 MB
manell/wireshark       latest              283b245afa45        5 months ago        574 MB
nginx                  alpine              935bd7bf8ea6        5 months ago        54.8 MB
memcached              alpine              fa6bdd4b792b        5 months ago        6.93 MB
node                   latest              800da22d0e7b        5 months ago        651 MB
fedora                 latest              11a5107645d4        5 months ago        204 MB
alpine                 latest              4e38e38c8ce0        7 months ago        4.8 MB
tutum/mongodb          latest              64ca9521c703        11 months ago       503 MB
[root@localhost basic]# docker rmi 64ca9521c703 800da22d0e7b 283b245afa45 e6400cb52bf1
Error response from daemon: reference does not exist
Error response from daemon: reference does not exist
Error response from daemon: reference does not exist
Error response from daemon: reference does not exist
[root@localhost basic]#
```

**Comment by mbenhalima** [comment]
To fix: `Error response from daemon: reference does not exist`

I had to delete the image from the filesystem:
` rm -rf /var/lib/docker/image/overlay2/imagedb/content/sha256/53912975086f1470f877922e3eb79c4f17c403fcb0b51b1f45df9b623987600e`

*** this was an image installed 4 years ago that i couldn't delete no matter what else i tried.

before:
```
# docker image ls
REPOSITORY    TAG       IMAGE ID       CREATED        SIZE
postgres      latest    53912975086f   4 years ago    312MB
```

```
After:
# docker image ls
REPOSITORY   TAG       IMAGE ID   CREATED   SIZE
#
```
